src/joy_api.c

```diff
@@ -109,75 +109,137 @@ static unsigned int joy_splt_format_data(flow_record_t *rec,
+                joy_timer_sub(&rec->pkt_time[i-1], &rec->pkt_time[entry_counter-1], &ts);
+                *(formatted_data+entries_used+entry_counter) =
+                     (uint16_t)joy_timeval_to_milliseconds(ts);
+                ++entry_counter;
             }
-        }
 
-        /* loop through the SPLT times and store appropriately */
-        for (i=0; i < num_of_pkts; ++i) {
+            /* write out the pkt length data */
+            *(formatted_data+entry_counter) = (uint16_t)rec->pkt_len[i];
+
+            /* calculate the time delta */
             if (i > 0) {
                 joy_timer_sub(&rec->pkt_time[i], &rec->pkt_time[i-1], &ts);
             } else {
                 joy_timer_sub(&rec->pkt_time[i], &rec->start, &ts);
             }
-            *(formatted_data+MAX_NFV9_SPLT_SALT_PKTS+i) =
+
+            /* write out the time data */
+            *(formatted_data+entries_used+entry_counter) =
                  (uint16_t)joy_timeval_to_milliseconds(ts);
-        }
 
-        if (num_of_pkts < MAX_NFV9_SPLT_SALT_PKTS) {
-            /* padding needs to occur */
-            for (;i < MAX_NFV9_SPLT_SALT_PKTS; ++ i) {
-               *(formatted_data+MAX_NFV9_SPLT_SALT_PKTS+i) = (uint16_t)0x00;
-            }
+            /* update our counters */
+            ++entry_counter;
+            last_len = rec->pkt_len[i];
+            run_count = 0;
+        } else {
+            /* found a run, keep track of it */
+            --run_count;
         }
+    }
 
-    /* else format for IPFix export */
-    } else {
-        /* loop through the SPLT lengths and store appropriately */
-        for (i=0; i < num_of_pkts; ++i) {
-            *(formatted_data+i) = (uint16_t)rec->pkt_len[i];
-        }
+    /* finished processing packets and a run was at the end */
+    if (run_count < 0) {
+        /* write out the run count */
+        *(formatted_data+entry_counter) = (int16_t)run_count;
 
-        /* store how many entries we used since IPFix doesn't pad */
-        entries_used = i;
+        /* write out the time delta for the run */
+        joy_timer_sub(&rec->pkt_time[i-1], &rec->pkt_time[i+run_count-1], &ts);
+        *(formatted_data+entries_used+entry_counter) =
+             (uint16_t)joy_timeval_to_milliseconds(ts);
 
-        /* loop through the SPLT times and store appropriately */
-        for (i=0; i < num_of_pkts; ++i) {
-            if (i > 0) {
-                joy_timer_sub(&rec->pkt_time[i], &rec->pkt_time[i-1], &ts);
-            } else {
-                joy_timer_sub(&rec->pkt_time[i], &rec->start, &ts);
+        /* update entry counter */
+        ++entry_counter;
+    }
+
+    /* pad out Netflow v9 export */
+    if (export_frmt == JOY_NFV9_EXPORT) {
+        if (entry_counter < MAX_NFV9_SPLT_SALT_PKTS) {
+            /* padding needs to occur */
+            for (i = entry_counter; i < MAX_NFV9_SPLT_SALT_PKTS; ++i) {
+               *(formatted_data+i) = (uint16_t)-32768;
```

Comment:
Since this is uint16 can we save this as a hex value?

Reply:
this value comes from the specification - I want it to match the specification. it is listed in 10 base integer format in the document.